### ISSUE TITLE
Specifying SSH key for AWS provider is not used

### ISSUE THREAD

**Issue by RunningJon** [yugabyte/yugabyte-db#7432]
The option to specify an existing SSH key for an AWS config is not used. The username is correctly used but the ssh key name and private key supplied are ignored. I confirmed the username is working by creating an AMI with the cloud user of "opc" for Oracle Enterprise Linux testing. This works but the ssh key name isn't used.

I confirmed his problem using 2.5.2 and specified my existing SSH key. The VMs created for the universe use a keypair created by Platform and you can see this with describe-instances or in the AWS console. 

The UI should also allow a user to specify the username but still let Yugabyte manage the keys. The username could be any number of values from centos, ec2-user, opc, or any other default username a customer may use in their AMI.

**Comment by streddy-yb** [comment]
hi @RunningJon - Can you confirm the steps you followed? It seems like you selected "Provide custom keypair information". Did you enter the custom keypair name / pem key etc? 

I created #7536 to track the ability to specify the username part. This issue will be used to address custom key-pair not being used.

![image](https://user-images.githubusercontent.com/68868569/110286764-4e488a00-7f9a-11eb-9994-a9faa23b8159.png)

**Comment by RunningJon** [comment]
Yes, I picked "Provide custom KeyPair information" and then picked an existing Keypair name and uploaded the PEM config file. It was ignored.

Also, if the AMI being used has a different cloud-init user other than centos, you have to pick customer KeyPair information so the SSH User parameter is exposed. The SSH User parameter should be tied to the AMI, not the KeyPair.

**Comment by RP1807** [comment]
Validated this issue on build 2.7.1.0-b184 with following steps
 
1. For AWS provider, selected "Provide custom KeyPair Information" option in KeyPair management and tried to submit without uploading a .pem file. UI throws a validation error as .pem file is mandatory. 
2. Created AWS provider with custom key pair without custom key pair name. Configured universe using this provider and tried to connect a node in the universe. Validated that connection is successful. 
3. Created AWS provider with custom key pair with custom key pair name. Configured universe using this provider and tried to connect to a node in the universe. Validated that connection is successful and Also UI uses custom key pair name in connect command. 
4. Deleted universe and corresponding aws provider, validated that the keys directory no longer has a subdir for the deleted provider. 

cc @VijiYB
